apps/passport-server/src/services/generic-issuance/pipelines/PretixPipeline.ts

```diff
@@ -7,37 +52,723 @@ import {
   PipelineType
 } from "./types";
 
+const LOG_NAME = "PretixPipeline";
+const LOG_TAG = `[${LOG_NAME}]`;
+
+/**
+ * This object represents a configuration from which the server can instantiate
+ * a functioning {@link PretixPipeline}. It's entirely specified by the user.
+ */
+export interface PretixPipelineOptions {
+  pretixAPIKey: string;
+  pretixOrgUrl: string;
+  events: PretixEventConfig[];
+}
+
 /**
- * TODO: implement this. (Probably Rob).
+ * Configuration for a specific event, which is managed under the organizer's
+ * Pretix account.
+ */
+export interface PretixEventConfig {
+  externalId: string; // Pretix's event ID
+  genericIssuanceId: string; // Our UUID
+  name: string; // Display name for the event
+  products: PretixProductConfig[];
+}
+
+/**
+ * Configuration for specific products available for the event. Does not need
+ * to include all products available in Pretix, but any product listed here
+ * must be available in Pretix.
+ */
+export interface PretixProductConfig {
+  externalId: string; // Pretix's item ID
+  genericIssuanceId: string; // Our UUID
+  name: string; // Display name
+  isSuperUser: boolean; // Is a user with this product a "superuser"?
+  // Superusers are able to check tickets in to events.
+}
+
+/**
+ * Class encapsulating the complete set of behaviors that a {@link Pipeline} which
+ * loads data from Pretix is capable of.
  */
 export class PretixPipeline implements BasePipeline {
   public type = PipelineType.Pretix;
   public capabilities = [
-    // TODO: fill this out with an issuance and checkin capability
+    {
+      issue: this.issuePretixTicketPCDs.bind(this),
+      feedId: "ticket-feed",
+      type: PipelineCapability.FeedIssuance,
+      getFeedUrl: (): string => generateIssuanceUrlPath(this.id)
+    } satisfies FeedIssuanceCapability,
+    {
+      checkin: this.checkinPretixTicketPCDs.bind(this),
+      type: PipelineCapability.Checkin,
+      getCheckinUrl: (): string => generateCheckinUrlPath(this.id)
+    } satisfies CheckinCapability
   ];
 
+  /**
+   * Used to sign {@link EdDSATicketPCD}
+   */
   private eddsaPrivateKey: string;
   private definition: PretixPipelineDefinition;
-  private db: IPipelineAtomDB;
+
+  /**
+   * This is where the Pipeline stores atoms so that they don't all have
+   * to be stored in-memory.
+   */
```

Comment:
as discussed keeping them in memory for now seems fine.

if that's the case, the `atom` abstraction isn't really that necessary imo. not opposed to removing it. not a huge priority tho.

Reply:
This comment was a copy-and-paste, but yes agreed.